mycpp/util.py

```diff
@@ -14,8 +14,33 @@
 
 SMALL_STR = False
 
+SymbolPath = tuple[str]
+
 
 def log(msg: str, *args: Any) -> None:
     if args:
         msg = msg % args
     print(msg, file=sys.stderr)
+
+def join_name(parts: SymbolPath, strip_package: bool = False, delim: str = '::') -> str:
```

Comment:
I think it's better not to have the param if we don't use it, if it's always ::

It can be a constant in the file, or just repeated

Reply:
Another PR I have sets this to `.` for use with the datalog files